### ISSUE TITLE
Supporting mod stores other than Thunderstore

### ISSUE THREAD

**Issue by Jan200101** [R2Northstar/Northstar#537]
With this issue I propose supporting mod.io as an alternative to the thunderstore.

**Comment by GeckoEidechse** [comment]
It seems mod.io doesn't want us as per @EM4Volts

> Hi `@4V`, we only support official integrations with studios and publishers, rather than unofficial mod communities. You could try moddb.com or gamefront.com as alternative sites we are affiliated with that do support such communities.

![image](https://github.com/R2Northstar/Northstar/assets/40122905/033c40f2-c464-46b2-99fd-d419f38696bb)

https://discord.com/channels/920776187884732556/942391932137668618/1139518734755709058


&nbsp;

Overall, I'm not opposed to switching mod platform but a proper argument needs to be made as it would basically require both yet another overhaul of the mod system in Northstar, an update to all existing mod managers, and deprecation of 1.5 years of documentation. All of that is a lot of overhead that requires a solid argument to be made for switching or enabling additional platforms.

**Comment by EM4Volts** [comment]
my 50 cents on this here:

first of all to adress something said already i quote you @GeckoEidechse 

> Overall, I'm not opposed to switching mod platform but a proper argument needs to be made as it would basically require both yet another overhaul of the mod system in Northstar, an update to all existing mod managers, and deprecation of 1.5 years of documentation. 

yes, mod managers would need an overhaul but there is essentially already a mod format in form of the way we have:
<author.mod> > folder shitt and mod.json

the 1.5 years of docs dont need to be deprecated as the only thing outdated would be how to make the manifest for thunderstore, which wouldnt be needed on other platforms that are just WAY simpler to just press an uplaod button.

so here come my arguments for it:

- Better moderation, even if there currently are just a couple bad mods (literal cheats, nsfw mods disguised as lgbtq stuff) its important that this exists

- Better upload system for modders, i have yet to see a plattform, be it for modding or even sharing cooking recipes with an upload/post system as bad as thunderstore, the fact that you have to premake the zip in a specific format for the site to accept it is just a HUGE mess.

- Actual feedback for modders, thunderstore has a thumbs up button that is as big as a single pixel on a phone screen, no comments nothing. its basically unused, theres mods that are months old with 1 upvote.

- Better filters, yet again a thing thunderstore seems horrible at.

- proper descriptions for mods, thunderstore gives u a literal github readme which most modders will not spend effort to make, other platforms allow u to properly make a good looking description, with a good on site editor for it.

these are the arguments for a new site that are purely based on my problems with thunderstore.
next up the stuff that could become even better:

You, @GeckoEidechse talked about "yet another overhaul of the mod system in Northstar", think about it that way, in theory this could lead to a new, better way too. imagine if this could be taken as a reason to actually invent a good PROPER universal system for managers to follow, not just some folder structure, installscripts and others possibly.

Users could find mods and dependencies easier, reducing tickets, and thus trouble.

Modders could get feedback without having to rely on discord which you yourself say is a bad plattform for discussion.



in my honest opinion i do not believe that the future of any project like northstar that isnt an big established title can be properly managed with a site like thunderstore unless BIG changes happen to the site, which personally, on recent blogs and more i never see coming.

**Comment by EM4Volts** [comment]
Extra to my previous comment: 
the way malicious mods are currently handled (pinging gecko, gecko telling the admins, the admins deleting it in about 30 years) is not a way anything should be handled

**Comment by GeckoEidechse** [comment]
> yes, mod managers would need an overhaul but there is essentially already a mod format in form of the way we have:
<author.mod> > folder shitt and mod.json

Keep in mind that for auto-updating with Thunderstore or any platform we still have to keep a mapping from that Northstar mod format to the mod on the external platform. Also there's also the fact that `author.mod` is not enforced in any ways and a lot of mods don't adhere to it at all, e.g. https://thunderstore.io/package/download/Creamy_jpg/Customisable_Reticles_Framework/1.3.6/ (4th most downloaded mod on Northstar Thunderstore).

Then there's also the fact that rn all user facing tutorials explain how to download things from Thunderstore, that what I meant also needs updating.


&nbsp;

I fully agree with the points made in the above btw. So I suggest looking into alternatives and seeing whether the feature set they offer would be worth switching to them.

My main point is that switching mod platform inherently is a lot of work as it will require updating all parts of Northstar. It's certainly not something you can do quickly on a rainy Friday night.

Hence step one would be exploring all viable alternatives and seeing whether they offer what we would need/want. We cannot afford attempting to switch just to find out that something crucial is missing.


&nbsp;

Also re:

> the way malicious mods are currently handled (pinging gecko, gecko telling the admins, the admins deleting it in about 30 years) is not a way anything should be handled

Just to be clear, I'm not the only person that can ping them. @H0L0theBard already did way before me. We're still awaiting response though.

**Comment by EM4Volts** [comment]
> > yes, mod managers would need an overhaul but there is essentially already a mod format in form of the way we have:
> > <author.mod> > folder shitt and mod.json
> 
> Keep in mind that for auto-updating with Thunderstore or any platform we still have to keep a mapping from that Northstar mod format to the mod on the external platform. Also there's also the fact that `author.mod` is not enforced in any ways and a lot of mods don't adhere to it at all.

yes i know, the autor.mod line was just an example for a folder name.

**Comment by uniboi** [comment]
What is the benefit of splitting up mods on yet another platform

**Comment by Jan200101** [comment]
Benefits:
- not having to deal with Thunderstores poor packaging system
- possibility for better moderation
- better community integration (comments, etc.)
- possible better in-game integration

**Comment by GeckoEidechse** [comment]
> - possible better in-game integration

Can also be done with Thunderstore and we kinda started the process for that with https://github.com/R2Northstar/NorthstarLauncher/pull/503 and #472

(other points are very valid ofc)

**Comment by Jan200101** [comment]
> Can also be done with Thunderstore

Thunderstores packaging structure does not lend itself to it.

**Comment by GeckoEidechse** [comment]
> Thunderstores packaging structure does not lend itself to it, also I said better no_mouth

Thunderstore packaging structure is undefined and specified by community. We just pinky-swear agreed to do `mods/` and `plugins/` subdirs to support integration with existing Northstar mods. So this can be easily changed without requiring any changes on Thunderstore side, only on Northstar side. The only thing that would need change is r2mm which we don't recommend using anyway.

**Comment by uniboi** [comment]
> not having to deal with Thunderstores poor packaging system

It's really not that bad

> possibility for better moderation

they're working on it

> better community integration (comments, etc.)

they're working on it

> possible better in-game integration

???

> Also tf you mean "yet another" 🤨

there are:
- 7(!) discord channels - where *not* everyone is also uploading on thunderstore
- thunderstore - which is integrated with every installer and northstar itself
- nexusmods - for some reason

**Comment by GeckoEidechse** [comment]
> - nexusmods - for some reason

Just to be clear, I have no idea who set up nexusmods and I wouldn't count it as an "officially supported" resource. I'd be more than happy to get rid of Discord channels but last time I tried the vote was in favour of keeping them.
(The Discord channel part is a separate discussion anyway though)

**Comment by uniboi** [comment]
Still, there is no point in having **multiple** "officially supported" mod stores, because I can guarantee that not everyone is going to upload to both of them.
And having two different standards is never a good idea

**Comment by EM4Volts** [comment]
i just talked to mythic, an admin of thunderstore via dm on discord and according to him both a better UI and proper on community basis moderation are in the works, moderation being actually not far away.

**Comment by Jan200101** [comment]
Got in contact with mod.io and they re-confirmed that unofficial modding support in any capacity is unsupported by mod.io and they are only intending to work with official products.

I've also asked how another game is on mod.io, which is just a decompile of a game with custom content support slapped on, and may have gotten it into trouble. Woops.

Either way, alternatives to mod.io that I see are:
- nexus
- moddb

nexus has been hostile towards modders for a while and moddb is rather lackluster for what we want.

So thunderstore continues being our best solution.
Once Thunderstore receives their UI overhaul we could consider contributing to it to implement what we want (e.g. improving the manual mod uploading process).

Since we've established that there are no other mod stores that we can reasonable support, this will be closed.

**Comment by EM4Volts** [comment]
reopening this, ive just spoken to someone from the team of https://modworkshop.net

Mod workshop.net is a modding website that has a nice modern design and basically all features one could want for a modding community. it has proper rules for uploads and allows for moderation by apointed mods of a community.

it has an api https://api.modworkshop.net/
but from what i heard it sadly doesnt allow for circumventing the need to go onto the page itself (which imo isnt a big issue as managers could propably also just add an way to just drag the downloaded zip into the manager), the api allows for updating existing mods tho

the people of their team told me "Hi, 3rd party frameworks are perfectly fine (Payday 2's mod loader is basically the same thing) as long as they don't actively go against the rules (ours as well as the developers of the game)."

so n* would be allowed with them.

**Comment by EM4Volts** [comment]
i also have a ticket open on their discord, if anyone wants to question them about something directly join their discord linked on their page and tell me, i can add you to the ticket
